src/algebra/pointwise.lean

```diff
@@ -789,10 +792,165 @@ lemma mul_singleton_zero [mul_zero_class α] (s : finset α) :
   s * {0} ⊆ {0} :=
 by simp [subset_iff, mem_mul]
 
-lemma singleton_zero_mul [mul_zero_class α] (s : finset α):
+lemma singleton_zero_mul [mul_zero_class α] (s : finset α) :
   {(0 : α)} * s ⊆ {0} :=
 by simp [subset_iff, mem_mul]
 
+end decidable_eq
+
+open_locale pointwise
+variables {α : Type*} {β : Type*} {s s₁ s₂ t t₁ t₂ u : finset α} {a b : α} {x y : β}
+
+/-- The finset `(1 : finset α)` is defined as `{1}` in locale `pointwise`. -/
+@[to_additive
+/-"The finset `(0 : finset α)` is defined as `{0}` in locale `pointwise`. "-/]
+protected def has_one [has_one α] : has_one (finset α) := ⟨{1}⟩
+
+localized "attribute [instance] finset.has_one finset.has_zero" in pointwise
+
+@[to_additive]
+lemma singleton_one [has_one α] : ({1} : finset α) = 1 := rfl
+
+@[simp, to_additive]
+lemma mem_one [has_one α] : a ∈ (1 : finset α) ↔ a = 1 :=
+by simp [has_one.one]
+
+@[to_additive]
+lemma one_mem_one [has_one α] : (1 : α) ∈ (1 : finset α) := by simp [has_one.one]
+
+@[simp, to_additive]
+theorem one_subset [has_one α] : 1 ⊆ s ↔ (1 : α) ∈ s := singleton_subset_iff
+
+@[to_additive]
+theorem one_nonempty [has_one α] : (1 : finset α).nonempty := ⟨1, one_mem_one⟩
+
+@[simp, to_additive]
+theorem image_one [decidable_eq β] [has_one α] {f : α → β} : image f 1 = {f 1} :=
+image_singleton f 1
+
+@[to_additive add_image_prod]
+lemma image_mul_prod [decidable_eq α] [has_mul α] :
+  image (λ x : α × α, x.fst * x.snd) (s.product t) = s * t := rfl
+
+@[simp, to_additive]
+lemma image_mul_left [decidable_eq α] [group α] :
+  image (λ b, a * b) t = preimage t (λ b, a⁻¹ * b) (assume x hx y hy, (mul_right_inj a⁻¹).mp) :=
+begin
+  have := @set.image_mul_left _ (t : set α) a _,
+  rw ← finset.coe_preimage _ (assume x hx y hy, (mul_right_inj a⁻¹).mp) at this,
+  exact_mod_cast this,
+end
```

Comment:
Can you get here faster using `equiv.mul_left` and some statement about taking the preimage by an equiv?

Reply:
I guess thats what the proof of the original image_mul_left is doing. I just kinda liked the proofs which are just saying, "hey this is a fact about sets really"

Reply:
Sure, but I think we have the set fact as `equiv.image_eq_preimage`, so the "hey this is a fact about sets really" would be within `equiv.finset_image_eq_preimage` or similar, where it's clear it's a fact about sets and not just the multiplication map.

Reply:
I couldn't really find a way to do this without adding a lot of lemmas, the proof is only 3 lines as it is so unless you have a specific suggestion that works I propose to leave this as is for this PR.

Reply:
It turns out that `finset.coe_injective $ by simp` works.